scipy/signal/signaltools.py

```diff
@@ -892,41 +892,84 @@ def _prod(iterable):
     return product
 
 
-def _fftconv_faster(x, h, mode):
+def _conv_ops(x_shape, h_shape, mode):
+    x_size, h_size = _prod(x_shape), _prod(h_shape)
+    if mode == "full":
+        out_shape = [n + k - 1 for n, k in zip(x_shape, h_shape)]
+    elif mode == "valid":
+        out_shape = [abs(n - k) + 1 for n, k in zip(x_shape, h_shape)]
+    elif mode == "same":
+        out_shape = x_shape
+    else:
+        raise ValueError("Acceptable mode flags are 'valid',"
+                         " 'same', or 'full', not mode={}".format(mode))
+
+    s1, s2 = x_shape, h_shape
+    if len(x_shape) == 1:
+        s1, s2 = s1[0], s2[0]
+        if mode == "full":
+            direct_ops = s1 * s2
+        elif mode == "valid":
+            direct_ops = (s2 - s1 + 1) * s1 if s2 >= s1 else (s1 - s2 + 1) * s2
+        elif mode == "same":
+            direct_ops = s1 * s2 if s1 < s2 else s1 * s2 - (s2 // 2) * ((s2 + 1) // 2)
+    else:
+        if mode == "full":
+            direct_ops = min(_prod(s1), _prod(s2)) * _prod(out_shape)
+        elif mode == "valid":
+            direct_ops = min(_prod(s1), _prod(s2)) * _prod(out_shape)
+        elif mode == "same":
+            direct_ops = _prod(s1) * _prod(s2)
+    fft_ops = sum(n * np.log(n) for n in (x_shape + h_shape + tuple(out_shape)))
+    return fft_ops, direct_ops
+
+
+def _get_fft_constant(mode, ndim, x_size, h_size, test=False):
+    if ndim == 1:
+        constants = {
+            "valid": 14.336458,
+            "full": 11.548068,
+            "same": 15.747428 if h_size <= x_size else 0.73367078,
```

Comment:
Presumably this should match as well
```suggestion
            "same": 15.747428 if h_size < x_size else 0.73367078,
```

Reply:
Thanks!